CMakeLists.txt

```diff
@@ -1,504 +1,2028 @@
-  set(dir "build")
-  set(cmd "cmake")
-  if (target)
-    set(dir "${dir}/${target}")
-    set(cmd "${cmd} -Dtarget=${target}")
-  elseif(CMAKE_BUILD_TYPE)
-    set(dir "${dir}/${CMAKE_BUILD_TYPE}")
-    set(cmd "${cmd} -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}")
-  else()
-    set(dir "${dir}/default")
-  endif()
-  set(cmd "${cmd} ${CMAKE_SOURCE_DIR}")
-
-  message(FATAL_ERROR "Builds are not allowed in ${CMAKE_SOURCE_DIR}.\n"
-    "Instead:\n"
-    "1) Remove the CMakeCache.txt file and CMakeFiles directory "
-   "from ${CMAKE_SOURCE_DIR}.\n"
-    "2) Create a directory to hold your build files, for example: ${dir}.\n"
-    "3) Change to that directory.\n"
-    "4) Run cmake targetting ${CMAKE_SOURCE_DIR}, for example: ${cmd}")
-endif()
-if("${CMAKE_GENERATOR}" MATCHES "Visual Studio" AND
-    NOT ("${CMAKE_GENERATOR}" MATCHES .*Win64.*))
-  message(FATAL_ERROR "Visual Studio 32-bit build is unsupported. Use
-    -G\"${CMAKE_GENERATOR} Win64\"")
-endif()
-
-setup_build_cache()
-
-if(nonunity)
-    get_cmake_property(allvars VARIABLES)
-    string(REGEX MATCHALL "[^;]*(DEBUG|RELEASE)[^;]*" matchvars "${allvars}")
-    foreach(var IN LISTS matchvars)
-        string(REGEX REPLACE "(DEBUG|RELEASE)" "\\1CLASSIC" newvar ${var})
-        set(${newvar} ${${var}})
-    endforeach()
-
-    get_cmake_property(allvars CACHE_VARIABLES)
-    string(REGEX MATCHALL "[^;]*(DEBUG|RELEASE)[^;]*" matchvars "${allvars}")
-    foreach(var IN LISTS matchvars)
-        string(REGEX REPLACE "(DEBUG|RELEASE)" "\\1CLASSIC" newvar ${var})
-        set(${newvar} ${${var}} CACHE STRING "Copied from ${var}")
-    endforeach()
-endif()
-
-determine_build_type()
-
-check_gcc4_abi()
-
-############################################################
-
-include_directories(
-  src
-  src/beast
-  src/beast/extras
-  src/nudb/include
-  src/soci/src
-  src/soci/include)
-
-special_build_flags()
-
-############################################################
-
-use_boost(
-  # resist the temptation to alphabetize these. coroutine
-  # must come before context.
-  chrono
-  coroutine
-  context
-  date_time
-  filesystem
-  program_options
-  regex
-  serialization
-  system
-  thread)
-
-use_pthread()
-
-use_openssl(${openssl_min})
-
-use_protobuf()
 
-setup_build_boilerplate()
+cmake_minimum_required (VERSION 3.9.0)
+set (CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/Builds/CMake ${CMAKE_MODULE_PATH})
+include (CMakeFuncs)
+include (CheckCXXCompilerFlag)
+if (target)
+  message (WARNING
+    "The target option is deprecated and will be removed in a future release")
+  parse_target()
+endif ()
+project (rippled)
+
+#[===================================================================[
+   convenience variables and sanity checks
+#]===================================================================]
+
+get_property (is_multiconfig GLOBAL PROPERTY GENERATOR_IS_MULTI_CONFIG)
+## this property is only in cmake 3.9+, so have a fallback/best-guess
```

Comment:
Since you changed the cmake min required version from 3.7 to 3.9, I think we can simplify this to not need the fallback.

Reply:
@mellery451 here's a commit you can pick if you want that should be safe: https://github.com/JoeLoser/rippled/commit/f39f9df183d0661793409ac5dcb7b61dd5dd9af2